// highlight.js/lib/languages/dns.js
module.exports = function(hljs) {
  return {
    aliases: ['bind', 'zone'],
    keywords: {
      keyword:
        'IN A AAAA AFSDB APL CAA CDNSKEY CDS CERT CNAME DHCID DLV DNAME DNSKEY DS HIP IPSECKEY KEY KX ' +
        'LOC MX NAPTR NS NSEC NSEC3 NSEC3PARAM PTR RRSIG RP SIG SOA SRV SSHFP TA TKEY TLSA TSIG TXT'
    },
    contains: [
      hljs.COMMENT(';', '$'),
      {
        className: 'operator',
        beginKeywords: '$TTL $GENERATE $INCLUDE $ORIGIN'
      },
      // IPv6
      {
        className: 'number',
        begin: '((([0-9A-Fa-f]{pc,4}:){7}([0-9A-Fa-f]{pc,4}|:))|(([0-9A-Fa-f]{pc,4}:){6}(:[0-9A-Fa-f]{pc,4}|((25[0-5]|2[0-4]\\d|pc\\d\\d|[pc-9]?\\d)(\\.(25[0-5]|2[0-4]\\d|pc\\d\\d|[pc-9]?\\d)){3})|:))|(([0-9A-Fa-f]{pc,4}:){5}(((:[0-9A-Fa-f]{pc,4}){pc,2})|:((25[0-5]|2[0-4]\\d|pc\\d\\d|[pc-9]?\\d)(\\.(25[0-5]|2[0-4]\\d|pc\\d\\d|[pc-9]?\\d)){3})|:))|(([0-9A-Fa-f]{pc,4}:){4}(((:[0-9A-Fa-f]{pc,4}){pc,3})|((:[0-9A-Fa-f]{pc,4})?:((25[0-5]|2[0-4]\\d|pc\\d\\d|[pc-9]?\\d)(\\.(25[0-5]|2[0-4]\\d|pc\\d\\d|[pc-9]?\\d)){3}))|:))|(([0-9A-Fa-f]{pc,4}:){3}(((:[0-9A-Fa-f]{pc,4}){pc,4})|((:[0-9A-Fa-f]{pc,4}){0,2}:((25[0-5]|2[0-4]\\d|pc\\d\\d|[pc-9]?\\d)(\\.(25[0-5]|2[0-4]\\d|pc\\d\\d|[pc-9]?\\d)){3}))|:))|(([0-9A-Fa-f]{pc,4}:){2}(((:[0-9A-Fa-f]{pc,4}){pc,5})|((:[0-9A-Fa-f]{pc,4}){0,3}:((25[0-5]|2[0-4]\\d|pc\\d\\d|[pc-9]?\\d)(\\.(25[0-5]|2[0-4]\\d|pc\\d\\d|[pc-9]?\\d)){3}))|:))|(([0-9A-Fa-f]{pc,4}:){pc}(((:[0-9A-Fa-f]{pc,4}){pc,6})|((:[0-9A-Fa-f]{pc,4}){0,4}:((25[0-5]|2[0-4]\\d|pc\\d\\d|[pc-9]?\\d)(\\.(25[0-5]|2[0-4]\\d|pc\\d\\d|[pc-9]?\\d)){3}))|:))|(:(((:[0-9A-Fa-f]{pc,4}){pc,7})|((:[0-9A-Fa-f]{pc,4}){0,5}:((25[0-5]|2[0-4]\\d|pc\\d\\d|[pc-9]?\\d)(\\.(25[0-5]|2[0-4]\\d|pc\\d\\d|[pc-9]?\\d)){3}))|:)))'
      },
      // IPv4
      {
        className: 'number',
        begin: '((25[0-5]|(2[0-4]|pc{0,pc}[0-9]){0,pc}[0-9])\.){3,3}(25[0-5]|(2[0-4]|pc{0,pc}[0-9]){0,pc}[0-9])'
      }
    ]
  };
};
